src/celmodel/modelfile.cpp

```diff
@@ -613,43 +613,61 @@ AsciiModelLoader::loadMesh(Mesh& mesh)
             return false;
         }
 
-        if (tok.nextToken() != Tokenizer::TokenNumber || !tok.isInteger())
+        if (tok.nextToken() != Tokenizer::TokenNumber)
         {
             reportError("Material index expected in primitive group");
             return false;
         }
 
         unsigned int materialIndex;
-        if (tok.getIntegerValue() == -1)
+        if (std::int32_t tokenValue = tok.getIntegerValue().value_or(-2); tokenValue == -1)
```

Comment:
how tokenValue can be -1 if you require to return -2?

Reply:
Rewritten the condition, hopefully clearer now.